ethcore/transaction/src/transaction/eip1559.rs

```diff
@@ -0,0 +1,166 @@
+        }
+        let list = Rlp::new(&d.as_raw()[1..]);
+        Ok(Eip1559Transaction {
+            chain_id: list.val_at(0)?,
+            nonce: list.val_at(1)?,
+            max_priority_fee_per_gas: list.val_at(2)?,
+            max_fee_per_gas: list.val_at(3)?,
+            gas: list.val_at(4)?,
+            action: list.val_at(5)?,
+            value: list.val_at(6)?,
+            data: list.val_at(7)?,
+            access_list: list.val_at(8)?,
+        })
+    }
+}
+
+/// Signed transaction information without verified signature.
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub struct UnverifiedEip1559Transaction {
+    /// Plain Transaction.
+    pub unsigned: Eip1559Transaction,
+    /// The V field of the signature
+    pub v: u64,
+    /// The R field of the signature; helps describe the point on the curve.
+    pub r: U256,
+    /// The S field of the signature; helps describe the point on the curve.
+    pub s: U256,
+    /// Hash of the transaction
+    pub hash: H256,
+}
+
+impl Deref for UnverifiedEip1559Transaction {
+    type Target = Eip1559Transaction;
+
+    fn deref(&self) -> &Self::Target { &self.unsigned }
+}
+
+impl rlp::Decodable for UnverifiedEip1559Transaction {
+    fn decode(d: &Rlp) -> Result<Self, DecoderError> {
+        let unsigned = Eip1559Transaction::decode(d)?;
+        let hash = keccak(d.as_raw());
+        let offset = unsigned.payload_length();
+        if d.as_raw().is_empty() {
+            return Err(DecoderError::RlpIsTooShort);
+        }
+        let list = Rlp::new(&d.as_raw()[1..]);
+        Ok(UnverifiedEip1559Transaction {
+            unsigned,
+            v: list.val_at(offset)?,
+            r: list.val_at(offset + 1)?,
+            s: list.val_at(offset + 2)?,
+            hash,
+        })
+    }
+}
+
+impl rlp::Encodable for UnverifiedEip1559Transaction {
+    fn rlp_append(&self, s: &mut RlpStream) { self.rlp_append_sealed_transaction(s) }
+}
+
+impl SignedTransactionShared for UnverifiedEip1559Transaction {
+    fn set_hash(&mut self, hash: H256) { self.hash = hash; }
+}
+
+impl UnverifiedEip1559Transaction {
+    /// tx list item count
+    fn payload_length(&self) -> usize { self.unsigned.payload_length() + 3 }
+
+    /// Append object with a signature into RLP stream
+    pub(crate) fn rlp_append_sealed_transaction(&self, s: &mut RlpStream) {
+        s.append(&(TxType::Type2 as u8));
+        s.begin_list(self.payload_length());
+        s.append(&self.chain_id);
+        s.append(&self.nonce);
+        s.append(&self.max_priority_fee_per_gas);
+        s.append(&self.max_fee_per_gas);
+        s.append(&self.gas);
+        s.append(&self.action);
+        s.append(&self.value);
+        s.append(&self.data);
+        s.append(&self.access_list);
+        s.append(&self.v);
+        s.append(&self.r);
+        s.append(&self.s);
+    }
+
+    pub fn standard_v(&self) -> u8 {
+        self.v.try_into().unwrap_or_default() // we should have parity 0 or 1  for v2 txns
```

Comment:
are you sure that `unwrap_or_default` wont have unexpected behavior?

Reply:
added v validation in [be513ae](https://github.com/KomodoPlatform/mm2-parity-ethereum/pull/1/commits/be513ae838fd5b6c3c425478fbe986511bbdebf2)

Reply:
Thanks